### PR TITLE
build: adding python3.11 support.

### PR DESCRIPTION
**Pull request by awais786**
on 3.11.9 they have made some changes which is breaking some tests. django will release its fix on 6th may.

Related [PR](https://github.com/python/cpython/commit/f97f25ef5dfcdfec0d9a359fd970abd139cf3428)
This is the root [cause](https://github.com/django/django/pull/17979#issuecomment-2061194558).

Upcoming fix.
https://github.com/django/django/pull/17979#issuecomment-2061194558

Only coverage is failing.

### CODE REVIEW

File: requirements/all.txt [hunk 1]
```diff
@@ -237,28 +228,14 @@ django-webpack-loader==3.0.1
...
     #   -r requirements/production.txt
     #   django-rest-swagger
     #   drf-jwt
     #   drf-yasg
     #   edx-drf-extensions
-docker[ssh]==7.0.0
-    # via
-    #   -r requirements/dev.txt
-    #   docker-compose
-docker-compose==1.29.2
```
Comment (awais786) [inline requirements/all.txt:252]: Need to move this to some where else. This is not compatible with `3.11`

**Comment by feanil** [comment]
@awais786 what's the status of this, are you still working on it?

File: requirements/all.txt (hunk 1 above)
Comment (iamsobanjaved) [reply requirements/all.txt:252]: This requirement isn't needed anymore as there is no usage left of docker-compose v1 in this repo.

**Comment by awais786** [comment]
So failures are coming due to 3.11.9 on 3.11.7 these tests are working fine.

**Comment by awais786** [comment]
@feanil fix will release on 6th may. Details [here](https://github.com/django/django/pull/17979#issuecomment-2061194558)

**Comment by awais786** [comment]
@feanil this seems valid issue. 
https://github.com/django/django/pull/17979#issuecomment-2061194558

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,11 +9,14 @@ on:
 jobs:
   quality_and_translations_tests:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.11.8"]
```
Comment (awais786) [inline .github/workflows/ci.yml:14]: on `3.11.9` they have made some changes which is breaking some tests. django will release its fix on 6th may.

**Comment by awais786** [comment]
Only coverage is failing due to some token issue.

**Comment by deborahgu** [comment]
> Only coverage is failing due to some token issue.

we know about the coverage failures, so not worried about that. Is this ready for review and (if it passes review) merge, in your opinion, @awais786? Or should we wait for 3.11.9 fix?

File: requirements/constraints.txt [hunk 3]
```diff
@@ -11,14 +11,14 @@
 # Common constraints for edx repos
 -c common_constraints.txt
 
-# The update to pyyaml 6.x failed as docker-compose wants <6,>=3.10. Pinning to <6.0. This constraint will be
```
[the remark this answers is not in the document]
Comment (feanil) [reply requirements/constraints.txt:14]: It looks like this was more of an issue of incompatibility with an old version of docker-compose but that package is no longer needed.

Pyyaml 6.x is compatible with Python 3.8 and 3.11

**Comment by feanil** [comment]
@awais786 the Django update is out, can you re-base and re-build requirements?

**Comment by awais786** [comment]
> @awais786 the Django update is out, can you re-base and re-build requirements?

I tried yesterday but i have lost push rights. @iamsobanjaved is working on this now.

**Comment by deborahgu** [comment]
I think I have permission to rebase and rebuild requirements, which I will do in a few minutes, since it's on me that I pulled in new  requirements before grabbing this branch.

**Review by deborahgu — APPROVED** [review]

**Pull request merged**